### PR TITLE
Add configuration for NousResearch Nous-Hermes-Llama2-13b model

### PR DESCRIPTION
**Pull request by m0saan**
### Description
This pull request adds a new configuration for the Nous-Hermes-Llama2-13b model from the NousResearch organization. The configuration includes parameters such as block size, vocabulary size, intermediate size, number of layers, heads, embeddings, query groups, and various other settings specific to the model.

### Changes Made
- Added a new configuration for the Nous-Hermes-Llama2-13b model.

### Related Issue
This PR addresses issue #427.

### CODE REVIEW

File: lit_gpt/config.py [hunk 1]
```diff
@@ -595,4 +595,29 @@ def norm_class(self) -> Type:
 configs.extend(freewilly_2)
 
 
+##########################
+# NousResearch Nous-Hermes-Llama2-13b
```
Comment (Andrei-Aksionov) [inline lit_gpt/config.py:599]: There is already `NousResearch` config section at line 482.
Could you put it there? And ideally to have the same order of arguments.

Comment (m0saan) [reply lit_gpt/config.py]: Yes sure!

File: lit_gpt/config.py [hunk 2]
```diff
@@ -496,7 +496,26 @@ def norm_class(self) -> Type:
...
+        name="Nous-Hermes-Llama2-13b",
+        block_size=4096,
+        vocab_size=32032,
+        n_layer=40,
+        n_head=40,
+        n_embd=5120,
+        rotary_percentage=1.0,
+        parallel_residual=False,
+        bias=False,
+        n_query_groups=40,
```
Comment (carmocca) [inline lit_gpt/config.py:513]: This isn't necessary

```suggestion
```

**Review by carmocca — COMMENTED** [review]
LGTM! Can you share some examples of generations with this checkpoint?

**Comment by m0saan** [comment]
> LGTM! Can you share some examples of generations with this checkpoint?

Is it acceptable to utilize the Transformers library for generating the examples?

**Comment by carmocca** [comment]
@m0saan You can follow the instructions described in https://github.com/Lightning-AI/lit-gpt/blob/main/tutorials/download_openllama.md but changing the checkpoint paths to use your newly added config

**Comment by m0saan** [comment]
@carmocca I followed the instruction you provided to generate an example with this current ckpt, but I am getting `torch.cuda.OutOfMemorvError: CUDA out of memory` error, also I checkout out this [tuto](https://github.com/Lightning-AI/lit-gpt/blob/main/tutorials/oom.md) but in vain.

Here is a screenshot of what I am getting!

Screenshot 2023-08-22 at 19 32 21 (2)

**Review by carmocca — COMMENTED** [review]
One thing that's missing for the NousResearch models is support for their prompt format: https://huggingface.co/NousResearch/Nous-Hermes-llama-2-7b#prompt-format

This would be added to https://github.com/Lightning-AI/lit-gpt/blob/main/chat/base.py#L204-L292

Do you think you can give it a shot?

File: lit_gpt/config.py [hunk 3]
```diff
@@ -496,7 +496,24 @@ def norm_class(self) -> Type:
         norm_eps=1e-6,
         _mlp_class="LLaMAMLP",
         intermediate_size=13824,
-    )
+    ),
+    # https://huggingface.co/NousResearch/Nous-Hermes-Llama2-13b
+    dict(
+        org="NousResearch",
+        name="Nous-Hermes-Llama2-13b",
```
[the remark this answers is not in the document]
Comment (m0saan) [reply lit_gpt/config.py:504]: done ✅ !

File: lit_gpt/config.py [hunk 4]
```diff
@@ -496,7 +496,42 @@ def norm_class(self) -> Type:
...
+        rotary_percentage=1.0,
+        parallel_residual=False,
+        bias=False,
+        _norm_class="RMSNorm",
+        norm_eps=1e-05,
+        _mlp_class="LLaMAMLP",
+        intermediate_size=13824,
+    ),
+    
+    dict(
```
Comment (Andrei-Aksionov) [inline lit_gpt/config.py:519]: Great job @m0saan !
But could you put `7b` before `13b` so they are listed in ascending order?

File: chat/base.py [hunk 5]
```diff
@@ -287,6 +287,14 @@ def prompt_config(checkpoint_dir: Path, tokenizer: Tokenizer) -> Tuple[str, Tupl
         )
         stop_tokens = ([tokenizer.eos_id],)
         return system_prompt, stop_tokens
+    
+    if re.search("NousResearch", checkpoint_name):
+        system_prompt = (
+            "### Instruction:\n{prompt}\n"
```
Comment (carmocca) [inline chat/base.py:293]: Doesn't the the recommended prompt format leave a newline in between?

```suggestion
            "### Instruction:\n{prompt}\n\n"
```

**Review by carmocca — APPROVED** [review]

File: lit_gpt/config.py [hunk 6]
```diff
@@ -480,6 +480,23 @@ def norm_class(self) -> Type:
 # NousResearch Hermes
 ######################
 nous_research = [
+    # https://huggingface.co/NousResearch/Nous-Hermes-llama-2-7b/blob/main/config.json
+    dict(
+        org="NousResearch",
+        name="Nous-Hermes-llama-2-7b",
```
Comment (carmocca) [inline lit_gpt/config.py:486]: I tested Nous-Hermes-Llama2-13 and it's working. I tried to test this checkpoint but I'm facing #371. After that's fixed, we can go ahead and merge this

**Comment by t-vi** [comment]
With #470  this should work. Thank you @carmocca for flagging it.

File: lit_gpt/config.py (hunk 6 above)
Comment (carmocca) [reply lit_gpt/config.py:486]: I had to change the vocab size. Now it works. Thanks!

**Pull request merged**